### PR TITLE
Process stderr

### LINKED ISSUE #92: Not all parallel jobs launching
When I go to run parallel inferences not all the chains start and then the call just hangs without ever returning.

This could just be my computer. I'll try to reproduce it at home, but I can get failures with something as simple as num_chains = 2, num_cores = 1, but it's easier to reproduce (the failures are random) with something like:

```
model$sample(num_chains = 10,
             num_cores = 10,
             data = data,
             save_warmup = 1,
             num_warmup = 100,
             num_sample = 0,
             metric = "dense_e",
             save_diagnostics = TRUE,
             term_buffer = 0,
             init_buffer = 100,
             window = 0)
```

As an example, output looks like:
```
...
Running ./accel_splines 'id=10' random 'seed=721749581' data 'file=/tmp/Rtmp2WZqkV/standata-17c1295a0363.json' output \
  'file=/tmp/Rtmp2WZqkV/accel_splines-stan-sample-10-17c12e1147ab.csv' \
  'diagnostic_file=/tmp/Rtmp2WZqkV/accel_splines-stan-sample-diagnostic-10-17c1457833ad.csv' 'method=sample' 'num_samples=0' \
  'num_warmup=100' 'save_warmup=1' 'algorithm=hmc' 'metric=dense_e' 'engine=nuts' adapt 'engaged=1' 'init_buffer=100' \
  'term_buffer=0' 'window=0'
Chain 4 Iteration:  1 / 100 [  1%]  (Warmup) 
Chain 5 Iteration:  1 / 100 [  1%]  (Warmup) 
Chain 8 Iteration:  1 / 100 [  1%]  (Warmup) 
Chain 9 Iteration:  1 / 100 [  1%]  (Warmup) 
Chain 10 Iteration:  1 / 100 [  1%]  (Warmup) 
Chain 10 Iteration: 100 / 100 [100%]  (Warmup) 
Chain 9 Iteration: 100 / 100 [100%]  (Warmup) 
Chain 10 finished in 3.1 seconds.
Chain 9 finished in 3.2 seconds.
Chain 4 Iteration: 100 / 100 [100%]  (Warmup) 
Chain 4 finished in 3.7 seconds.
Chain 5 Iteration: 100 / 100 [100%]  (Warmup) 
Chain 5 finished in 3.8 seconds.
Chain 8 Iteration: 100 / 100 [100%]  (Warmup) 
Chain 8 finished in 4.2 seconds.
```

Github wouldn't let me upload R scripts so I'm just putting the files on discourse: https://discourse.mc-stan.org/t/problems-running-multiple-chains-in-cmdstanr/11997

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and agree to license (see below)

#### Summary

Fixes #92 and might also squash #69 for good now.

Adds printing of stderr to the background process handling.

Thanks @bbbales2 for finding this one.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar, Univ. of Ljubljana

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
We might want to add additional arguments that would enable users to suppress warnings, though that might as well be a new issue and a separate pull as proper processing of stderr is more of an "urgent" feature.

**Comment by rok-cesnovar** [comment]
@bbbales2 can I ask you to double check for me if this fixes the case you explained in #69

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ cache:
 # FIXME: replace with cmdstanr master eventually
 r_github_packages:
   - r-lib/covr
-  - stan-dev/cmdstanr@install_cmdstan_from_r
+  - stan-dev/cmdstanr
```
Comment (bbbales2) [inline .travis.yml]: What's this line?

**Comment by bbbales2** [comment]
I'll merge it if it passes tests! Thanks!

File: .travis.yml (hunk 1 above)
Comment (rok-cesnovar) [reply .travis.yml]: Travis was looking for the wrong branch. Switched to master. Unrelated to this PR but travis will fail without this fix.

**Comment by jgabry** [comment]
All testing passed! Gonna go ahead and merge

**Pull request merged**